### ISSUE TITLE
Dynamic configuration

### ISSUE THREAD

**Issue by jbendes** [snsinfu/reverse-tunnel#20]
Is there any way we can change the agent config while the server is running?

**Comment by jbendes** [comment]
Yes that is exactly what I meant! Do you have any thoughts on how you'd want that implemented? I would be happy to submit a PR. This is an awesome awesome tool you've made

**Comment by snsinfu** [comment]
Thanks! PR is welcome.

I would trap `SIGHUP` signal (standard-ish signal for reloading config) in the main package and reconfigure the running server with a newly loaded configuration upon signal arrival.

In the current implementation, agent configurations are passed to [Action](https://github.com/snsinfu/reverse-tunnel/blob/v1.3.2/server/action.go) from [here](https://github.com/snsinfu/reverse-tunnel/blob/286fb18f077182e01665cc118d4785c07b042975/server/server.go#L34) and are used to create [tcp.Service](https://github.com/snsinfu/reverse-tunnel/blob/v1.3.2/server/tcp/service.go) and [udp.Service](https://github.com/snsinfu/reverse-tunnel/blob/v1.3.2/server/udp/service.go) objects (which handle agent auth). Luckily, these service objects are stateless, so we can replace them with new ones on the fly. I'd add `Action.Update(conf)` method or something to update tcp and udp services.

I'm not sure what to do with a change of `control_address`. Maybe the program should detect incompatible changes and refuse to reload a new config.

**Comment by berttejeda** [comment]
Is this ability also being implemented for the client component?
I'm interested in dynamically initializing connections from the client config.
